testing/test-cases.d/signer.support_229/test.sh

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env bash
+#
+# Regression test case for SUPPORT-229: Broken zone when ODS replaces the DNSKEY but does not re-sign RRsets using the new DNSKEY.
+# Specifically, after the ZSK <Lifetime> period but before the <Refresh> period, with the customers kasp.xml settings ODS replaces
+# the only ZSK DNSKEY (used to create the RRSIGs in the zone) with a new one but does NOT replace the RRSIGs, so the RRSIG records
+# are unusable because the DNSKEY they must be verified against no longer exists in the zone.
+#
+# TO DO: Is the missing DNSKEY lacking the <Publish/> stanza in the signconf? YES it is - augment the test to check this?
+# TO DO: Match RRSIGs against DNSKEYs using something better than relying on zone comments?
+# TO DO: Base the leap time on the actual kasp.xml ZSK <Lifetime> value, not a hard-coded leap time.
+# TO DO: Understand why it breaks after 31 days but not after 30 days.
+# TO DO: Refactor into more helper functions?
+# TO DO: Reduce to the simplest reproduction scenario, e.g. without enforcer(d) but only with signer(d) using constructed signconf.
+#
+SIGNED_ZONES="$INSTALL_ROOT/var/opendnssec/signed"
+SIGNCONF_FILE="${INSTALL_ROOT}/var/opendnssec/signconf/example.com.xml"
+TESTDIR=$(dirname $0)
+THIRTY_ONE_DAYS_IN_SECONDS=$((31*24*60*60))
+LIGHT_GREEN='\033[0;92m'
+NO_COLOUR='\033[0m'
+
+# helper functions
+l() {
+    echo -en "${LIGHT_GREEN}$(date -u +'%b %d %H:%M:%S') UTC: LINE $1: ${NO_COLOUR}"
```

Comment:
FreeBSD does not support -e
I would leave out the coloring, it might clutter the output when viewing from Jenkins